[limyuquan] iP

limyuquan opened the pull request:
# PR for grading :smile:

- Item 1
- Item 2
- Item 3

1. First item
2. Second item
3. Third item

```javascript
console.log("Hello, world!");
```

- [ ] Task 1
- [x] Task 2 (completed)
- [ ] Task 3

> This is a blockquote.

[My github profile](https://github.com/limyuquan)

`inline code`

*italic*
**bold**
~~strikethrough~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

limyuquan committed:
Complete Level-0

limyuquan committed:
Finished Level-1

limyuquan committed:
Finished Level-2

limyuquan committed:
Finished Level-3

limyuquan committed:
Finished Level-4

limyuquan committed:
Finished Level-5

limyuquan committed:
Finished Level-5

limyuquan committed:
Finished Level-6

limyuquan committed:
Finished A-Enums

limyuquan committed:
refactored code for reusability

limyuquan committed:
Implement Level 7: Add file reading and writing functionality

limyuquan committed:
Merge branch-Level-7 into master

limyuquan committed:
Implement Level 8: Add Date time support

limyuquan committed:
Merge branch-Level-8 into master

limyuquan committed:
Implement A-MoreOOP: refactor code as classes

limyuquan committed:
Implement A-MoreOOP: refactor code as classes

limyuquan committed:
Mege branch-A-MoreOOP into master

limyuquan committed:
Implement A-Packages: Divide Classes into packages

limyuquan committed:
Merge A-Packages into master

limyuquan committed:
merge add-gradle support into master

limyuquan committed:
add files to gitignore

limyuquan committed:
Implement Unit tests

limyuquan committed:
Merge branch-A-JUnit into master

limyuquan committed:
Change settings for JAR file output

limyuquan committed:
Merge A-Jar into main

limyuquan committed:
Add JavaDoc

limyuquan committed:
Fix code to follow coding standard

limyuquan committed:
Add Find Function

limyuquan committed:
Merge branch-Level-9 into master

limyuquan committed:
Merge coding standard into master

limyuquan committed:
Merge JavaDocs into master

limyuquan committed:
Added CheckStyle

limyuquan committed:
Implement GUI

limyuquan committed:
Merge branch-Level-10 into main

niclammm commented on src/test/java/chatkaki/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,21 @@
+public class DeadlineTest {
+
+    @Test
+    public void fileFormat_validDeadline_correctFormat() {
```
I think camelCase such as getName(), computeTotalWidth() should be used instead of snakeCase

niclammm commented on src/test/java/chatkaki/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    @Test
+    public void toString_validDeadline_correctString() {
```
Same here!

niclammm commented on src/main/java/chatkaki/ChatKaki.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     * Starts the chat service.
+     */
+    public static void startChatService() {
```
The startChatService() function does multiple things: it greets the user, handles user input, and manages command execution. Maybe you can consider breaking it down into separate methods for better readability. However, this allows the code to be concise which is good as well

niclammm commented on src/main/java/chatkaki/Storage.java:
```diff
@@ -0,0 +1,73 @@
+                        break;
+                }
+            }
+        } catch (Exception e) {
```
The catch blocks handle exceptions by printing the error message. This is a good start, but consider logging the errors or providing more context-specific error messages to help with debugging.

niclammm reviewed (commented):
Interesting approach and most of your code is accordance to the Java coding standard including Naming, Cases and more. Good job and keep up the good work!

limyuquan committed:
Add assert statements to ensure conditions are met

- The code currently lacks assert statements to verify certain conditions
- This can lead to potential issues going unnoticed during developement.
- Adding assert statements in various methods ensures that conditions are met,
improving code relaiability by catching potential issues early

limyuquan committed:
Merge pull request #1 from limyuquan/branch-A-Assertions

Add assert statements to ensure conditions are met

limyuquan committed:
Add styling and Improve Code Quality

- Currently the styling looks boring and unapitising, making user not really want to use it
- Currently there are some Code Quality issues with imports

limyuquan committed:
Merge pull request #2 from limyuquan/branch-A-CodeQuality

Add styling and Improve Code Quality

limyuquan committed:
Add extention

limyuquan committed:
Update GUI

limyuquan committed:
Merge pull request #3 from limyuquan/branch-A-BetterGui

Update GUI

limyuquan committed:
Add user guide

limyuquan committed:
Uncomment out large chunk

limyuquan committed:
Update Readme

limyuquan committed:
Remove deprecation

limyuquan committed:
Change version

limyuquan committed:
Remove unneccessary code

limyuquan committed:
Update jar version
